pyfstat/injection_parameters.py

```diff
@@ -1,83 +1,256 @@
     Draw injection parameter samples from priors and return in dictionary format.
 
     Parameters
     ----------
-    priors: dict
+    priors:
         Each key refers to one of the signal's parameters
         (following the PyFstat convention).
-        Priors can be given as values in three formats
-        (by order of evaluation):
 
-        1. Callable without required arguments:
-        `{"ParameterA": np.random.uniform}`.
+        Each parameter's prior should be given as a dictionary entry as follows:
+        `{"parameter": {"<function>": {**kwargs}}}`
+        where <function> may be (exclusively) either a user-defined function decorated
+        with `@custom_prior` or the name of a `scipy.stats` random variable.
+
+        * | If a user-defined function is used, such a function *must* take
+          | a `generator` kwarg as one of its arguments and use such a generator
+          | to generate any required random number within the function.
+          | The `generator` kwarg is *required* regardless of whether this is a
+          | deterministic or random function.
+          | For example, a negative log-distributed random number could be constructed as
+
+        ::
 
-        2. Dict containing numpy.random distribution as key and kwargs in a dict as value:
-        `{"ParameterA": {"uniform": {"low": 0, "high":1}}}`.
+            @pyfstat.custom_prior
+            def negative_log_uniform(generator):
+                return -10**(generator.uniform())
 
-        3. Constant value to be returned as is:
-        `{"ParameterA": 1.0}`.
+        * | If a `scipy.stats` function is used, it *must* be given as `stats.*`
+          | (i.e. the `stats` namespace should be explicitly included).
+          |  For example, a uniform prior between 3 and 5 would be written as
+          | `{"parameter": {"stats.uniform": {"loc": 3, "scale": 5}}}`.
 
-    seed: None, int
-        Argument to be fed to numpy.random.default_rng,
-        with all of its accepted types.
+        Delta-priors (i.e. priors for a determinisitic output) can also be specified by
+        giving the fixed value to be returned as-is. For example, specifying a fixed
+        value of 1 for the parameter `A` would be `{"A": 1}`.
 
-    _rng: np.random.Generator
-        Alternatively, this class accepts an already-initialized np.Generator,
-        in which case the `seed` argument will be ignored.
+        Alternatively, the following three options, which were recommended
+        on a previous release, are still a valid input. They will be used as a fall-back
+        if none of the two previous options are matched,
+        but their use is highly discouraged for newly produced code.
+
+            1. Callable without required arguments:
+            `{"ParameterA": np.random.uniform}`.
+
+            2. Dict containing numpy.random distribution as key and kwargs
+            in a dict as value:
+            `{"ParameterA": {"uniform": {"low": 0, "high":1}}}`.
```

Comment:
Confusing: sounds like the _syntax_ of that dict of dicts has changed, but actually only the expected contents are now different.